[python3:x64-windows] [windows sdk 10.0.22000.0] build failure

**Host Environment**
- OS: Windows 10
- Compiler: VS 2019 14.25.28610

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install python3:x64-windows`

**Failure logs** 
```
Microsoft (R)-Build-Engine, Version 16.5.0+d4cbfca49 für .NET Framework
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: msbuild D:/LocalGit/Repos/Microsoft/vcpkg/buildtrees/python3/x64-windows-rel/v3.9.2-cc9ebdcfd9.clean
/PCbuild/pcbuild.proj /p:Configuration=Release /p:IncludeExtensions=true /p:IncludeExternals=true /p:IncludeCTypes=true
/p:IncludeSSL=true /p:IncludeTkinter=false /p:IncludeTests=false /p:ForceImportBeforeCppTargets=D:/LocalGit/Repos/Micros
oft/vcpkg/buildtrees/python3/src/v3.9.2-cc9ebdcfd9.clean/PCbuild/python_vcpkg.props /p:IncludeUwp=false /p:_VcpkgPythonL
inkage=DynamicLibrary /t:Rebuild /p:Platform=x64 /p:PlatformToolset=v142 /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMK
L=No /p:WindowsTargetPlatformVersion=10.0.18362.0 /p:VcpkgTriplet=x64-windows /p:VcpkgInstalledDir=D:/LocalGit/Repos/Mic
rosoft/vcpkg/installed /p:VcpkgManifestInstall=false /m
    Working Directory: D:/LocalGit/Repos/Microsoft/vcpkg/buildtrees/python3/x64-windows-rel/v3.9.2-cc9ebdcfd9.clean
    Error code: 1
    See logs for more information:
      D:\LocalGit\Repos\Microsoft\vcpkg\buildtrees\python3\build-x64-windows-rel-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_msbuild.cmake:165 (vcpkg_execute_required_process)
  ports/python3/portfile.cmake:110 (vcpkg_install_msbuild)
  scripts/ports.cmake:142 (include)


Error: Building package python3:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: python3:x64-windows
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e
```
**Output-Log**
[build-x64-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/6291290/build-x64-windows-rel-out.log)

`error MSB8036: The Windows SDK version 7.0 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [E:\thirdPart\vcpkg\buildtrees\python3\x64-windows-rel\v3.9.2-cc9ebdcfd9.clean\PCbuild\pythoncore.vcxproj]
`

… And it failed again. Argh.

Log excerpt from the latest run:

```log
  Starting package 93/109: python3:x64-windows
  Building package python3[core]:x64-windows...
  -- Downloading https://github.com/python/cpython/archive/v3.9.7.tar.gz -> python-cpython-v3.9.7.tar.gz...
  -- Extracting source D:/a/Vega-Strike-Engine-Source/Vega-Strike-Engine-Source/v/downloads/python-cpython-v3.9.7.tar.gz
  -- Applying patch 0002-use-vcpkg-zlib.patch
  -- Applying patch 0003-devendor-external-dependencies.patch
  -- Applying patch 0004-dont-copy-vcruntime.patch
  -- Applying patch 0005-only-build-required-projects.patch
  -- Applying patch 0006-fix-duplicate-symbols.patch
  -- Using source at D:/a/Vega-Strike-Engine-Source/Vega-Strike-Engine-Source/v/buildtrees/python3/src/v3.9.7-237da30ffc.clean
  -- Building PCbuild/pcbuild.proj for Release
  CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
  Error:     Command failed: msbuild D:/a/Vega-Strike-Engine-Source/Vega-Strike-Engine-Source/v/buildtrees/python3/x64-windows-rel/v3.9.7-237da30ffc.clean/PCbuild/pcbuild.proj /p:Configuration=Release /p:IncludeExtensions=true /p:IncludeExternals=true /p:IncludeCTypes=true /p:IncludeSSL=true /p:IncludeTkinter=false /p:IncludeTests=false /p:ForceImportBeforeCppTargets=D:/a/Vega-Strike-Engine-Source/Vega-Strike-Engine-Source/v/buildtrees/python3/src/v3.9.7-237da30ffc.clean/PCbuild/python_vcpkg.props /p:WindowsTargetPlatformVersion=10.0.22000.0 /p:DefaultWindowsSDKVersion=10.0.22000.0 /p:IncludeUwp=false /p:_VcpkgPythonLinkage=DynamicLibrary /t:Rebuild /p:Platform=x64 /p:PlatformToolset=v142 /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /p:WindowsTargetPlatformVersion=10.0.22000.0 /p:VcpkgTriplet=x64-windows /p:VcpkgInstalledDir=D:/a/Vega-Strike-Engine-Source/Vega-Strike-Engine-Source/build/vcpkg_installed /p:VcpkgManifestInstall=false /m
```

So it's still using version 10.0.22000.0 of the Windows SDK here, even though I told CMake to use version 10.0.19041.0. Is there a way to change the version of the WinSDK used here as well?

--- a/ports/python3/0008-workaround-windows-11-sdk-rc-compiler-error.patch
+++ b/ports/python3/0008-workaround-windows-11-sdk-rc-compiler-error.patch
@@ -1,0 +1,30 @@
+From 0a72b7061ed79c5d6d37b41a5b1610e32fb371a4 Mon Sep 17 00:00:00 2001
+From: Adam Johnson <AdamJohnso@gmail.com>
+Date: Wed, 22 Sep 2021 21:04:21 -0400
+Subject: [PATCH] workaround windows 11 sdk rc compiler error
+
+winnt.h was changed to error if the `SYSTEM_CACHE_ALIGNMENT` cannot be
+determined. when the RC compiler is invoked, this seems to fail where
+previous SDKs silently set the ARM value.
+---
+ PC/python_ver_rc.h | 7 +++++++
+ 1 file changed, 7 insertions(+)
+
+diff --git a/PC/python_ver_rc.h b/PC/python_ver_rc.h
+index 90fc6ba1a1..e313a5138e 100644
+--- a/PC/python_ver_rc.h
++++ b/PC/python_ver_rc.h
+@@ -1,3 +1,10 @@
++// Temporarily workaround bug in Windows SDK 10.0.22000.0 winnt.h
++#ifdef RC_INVOKED
++#   ifndef SYSTEM_CACHE_ALIGNMENT_SIZE
++#       define SYSTEM_CACHE_ALIGNMENT_SIZE 64
++#   endif
++#endif
++
+ // Resource script for Python core DLL.
+ // Currently only holds version information.
+ //
+-- 
+2.33.0.windows.1
+

--- a/ports/python3/portfile.cmake
+++ b/ports/python3/portfile.cmake
@@ -28,6 +28,14 @@ elseif(VCPKG_TARGET_IS_WINDOWS AND CMAKE_SYSTEM_VERSION EQUAL 6.1)
     message(FATAL_ERROR "python3 requires the feature deprecated-win7-support when building on Windows 7.")
 endif()
 
+# The Windows 11 SDK has a problem that causes it to error on the resource files, so we patch that.
+if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
+    vcpkg_get_windows_sdk(WINSDK_VERSION)
+    if("${WINSDK_VERSION}" VERSION_GREATER_EQUAL "10.0.22000")
+        list(APPEND PATCHES "0008-workaround-windows-11-sdk-rc-compiler-error.patch")
+    endif()
+endif()
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO python/cpython
@@ -88,11 +96,6 @@ if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
             "/p:ForceImportBeforeCppTargets=${SOURCE_PATH}/PCbuild/python_vcpkg.props"
         )
     endif()
-    string(REPLACE "\\" "" WindowsSDKVersion "$ENV{WindowsSDKVersion}")
-    list(APPEND OPTIONS
-        "/p:WindowsTargetPlatformVersion=${WindowsSDKVersion}"
-        "/p:DefaultWindowsSDKVersion=${WindowsSDKVersion}"
-    )
     if(VCPKG_TARGET_IS_UWP)
         list(APPEND OPTIONS "/p:IncludeUwp=true")
     else()
@@ -117,6 +120,7 @@ if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
         PROJECT_SUBPATH "PCbuild/pcbuild.proj"
         OPTIONS ${OPTIONS}
         LICENSE_SUBPATH "LICENSE"
+        TARGET_PLATFORM_VERSION "${WINSDK_VERSION}"
         SKIP_CLEAN
     )
 

--- a/ports/python3/vcpkg.json
+++ b/ports/python3/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "python3",
   "version-semver": "3.9.7",
+  "port-version": 1,
   "description": "The Python programming language",
   "homepage": "https://github.com/python/cpython",
   "supports": "!(arm | uwp)",

--- a/ports/python3/vcpkg.json
+++ b/ports/python3/vcpkg.json
@@ -34,14 +34,6 @@
       "name": "sqlite3",
       "platform": "!(windows & static)"
     },
-    {
-      "name": "vcpkg-cmake",
-      "host": true
-    },
-    {
-      "name": "vcpkg-cmake-config",
-      "host": true
-    },
     "zlib"
   ],
   "features": {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5278,7 +5278,7 @@
     },
     "python3": {
       "baseline": "3.9.7",
-      "port-version": 0
+      "port-version": 1
     },
     "qca": {
       "baseline": "2.3.1",

--- a/versions/p-/python3.json
+++ b/versions/p-/python3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "6a1011c32aa2eb7de0b9a6184b2805e34c41f0b5",
+      "version-semver": "3.9.7",
+      "port-version": 1
+    },
+    {
       "git-tree": "b3a545b39c982b7f0a12891765dd9909e364ebc6",
       "version-semver": "3.9.7",
       "port-version": 0

